common/lib/dependabot/errors.rb

```diff
@@ -294,6 +311,31 @@ def replace_capture_groups(string, regex, replacement)
     end
   end
 
+  class TypedDependabotError < Dependabot::DependabotError
```

Comment:
Created to provided defining embedded error type in sub errors and provide function to get detail with the type.